dashboard/modules/metrics/metrics_head.py

```diff
@@ -217,9 +222,71 @@ def _create_default_prometheus_configs(self):
         os.makedirs(os.path.dirname(prometheus_config_output_path), exist_ok=True)
         shutil.copy(PROMETHEUS_CONFIG_INPUT_PATH, prometheus_config_output_path)
 
+    async def _check_grafana_running(self) -> bool:
+        try:
+            if self.grafana_host == GRAFANA_HOST_DISABLED_VALUE:
+                return False
+
+            path = f"{self.grafana_host}/{GRAFANA_HEALTHCHECK_PATH}"
+            async with self._session.get(path) as resp:
+                if resp.status != 200:
+                    return False
+
+                json = await resp.json()
+                # Basic sanity check of grafana health check schema
+                if "version" not in json:
+                    return False
+
+                return True
+        except Exception:
+            return False
+
+    async def _check_prometheus_health(self) -> bool:
+        try:
+            path = f"{self.prometheus_host}/{PROMETHEUS_HEALTHCHECK_PATH}"
+
+            async with self._session.get(path) as resp:
+                if resp.status != 200:
+                    return False
+
+                text = await resp.text()
+                # Basic sanity check of prometheus health check schema
+                if "Prometheus" not in text:
+                    return False
+
+                return True
+        except Exception:
+            return False
+
+    async def _push_telemetry_data_loop(self):
```

Comment:
In `usage_stats_head.py`, we have a loop to report usage stats periodically. We can push the dashboard related usage stats there so we don't need to have a different cadence.

Reply:
Interesting. Do we want to have `usage_stats_head` both report telemetry data _and_ emit the telemetry data?

In this current approach, it may be a different cadence but at least the metric is collocated with the component that cares most about that metric.

Reply:
Yeah the cluster-related data that can be obtained from the dashboard is generated from `usage_stats_head `. I think one potential solution is to call /grafana_health endpoint from usage_stats_head.py. Not sure if we can send a request to itself though

Reply:
yea `usage_stats_head` also collect (emit) usage stats besides reporting them so it should be fine. One benefit of the same cadence is it's easy for testing. For example, for testing, we change the cadence to every 1s but if the other cadence is still 10 minutes then we will miss the data.

Also in theory, we don't need to upload these two key-values to GCS, we can just get the values and add them directly to the data dict `usage_stats_head` will report. But we can also still follow the same pattern for consistency.

Reply:
Is there an existing pattern today of modules calling functions from other modules? Would like to re-use that pattern if possible.

Reply:
Actually I just realized that `metrics_head` is not minimal head which means if it's not loaded then we won't report at all? So seems we have three states: true, false, missing.

Reply:
is it possible to just ping the endpoint here? I think there are 2 reasons I prefer this approach.

1. It is probably the bad assumption the usage_head will co-exist with other dashboard modules. As Jiajun said metrics head doesn't exist in the minimal install. Or there's a possibility we will split this out to other component (when we split job submission server and the dashboard).
2. Currently, there's no call dependency between modules. I'd like to keep this behavior if possible